code/game/objects/items/tools/flame_tools.dm

```diff
@@ -170,7 +173,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	damtype = "brute"
 	icon_state = "[initial(icon_state)]_burnt"
 	item_state = "cigoff"
-	if(user && loc != user)
+	if(user)
 		user.SetLuminosity(0, FALSE, src)
```

Comment:
And then above here add 'var/mob/user = loc' while removing the arg from the proc